src/radio/radio-device-manager.ts

```diff
@@ -70,6 +70,47 @@ export class RadioDeviceManager implements smarthome.DeviceManager {
     );
   }
 
+  /**
+   * Fulfills an `HttpRequestData`.
+   * @param httpRequestData  The `HttpRequestData` to source radio parameters from.
+   * @returns  A promise that resolves to the determined `HttpResponseData`.
+   */
+  private async processHttpResponseData(
+    httpRequestData: smarthome.DataFlow.HttpRequestData
+  ): Promise<smarthome.DataFlow.HttpResponseData> {
+    const address = this.deviceIdToAddress.get(httpRequestData.deviceId)!;
+    let port = 80;
+    if (httpRequestData.port !== undefined) {
+      port = httpRequestData.port;
+    }
+    if (httpRequestData.method === 'GET') {
+      const httpResponse = await this.radioController.sendHttpGet(
```

Comment:
Is there a reason that this function returns an `HttpResponse` rather than directly creating an `HttpResponseData`?

Reply:
This probably relates to some of the discussion in https://github.com/googleinterns/smart-home-local/pull/48#discussion_r467096640.

The main reason is to abstract the radio fulfillment from platform logic, so that `RadioController` doesn't need to know about `requestId` or `deviceId`.  This way, we can swap out `RadioController` more easily (see `ProxyRadioClient` #56).